base/iterators.jl

```diff
@@ -1171,10 +1171,10 @@ function length(itr::PartitionIterator)
     return cld(l, itr.n)
 end
 
-function iterate(itr::PartitionIterator{<:AbstractRange}, state=1)
-    state > length(itr.c) && return nothing
+function iterate(itr::PartitionIterator{<:AbstractVector}, state = firstindex(itr.c))
```

Comment:
`PartitionIterator{<:AbstractVector}` would make all `AbstractRange` invalid so definitely not okay here.

Reply:
I'm not clear with "invalid", if we add a dispatch for `PartitionIterator{<:AbstractVector}` , then everything is the same except for replacing `view` with `getindex`.

Reply:
Because `UnitRange <: AbstractVector` is `false`, this change of method signature will make `PartitionIterator{<:UnitRange}` unable to match this method.

Anything wrong with the following version?

```suggestion
function iterate(itr::PartitionIterator{<:AbstractRange}, state = firstindex(itr.c))
```

Reply:
on 1.6.1
```julia
julia> UnitRange <: AbstractRange <: AbstractVector
true
```
something changed in dev?

Reply:
Oops. My fault. I didn't realize that `AbstractRange <: AbstractVector`. Sorry for the misleading.